wp-cli/i18n-command
# Extract strings for translation from theme.json

Closes https://github.com/wp-cli/i18n-command/issues/224

## Setup

- See https://make.wordpress.org/cli/handbook/contributions/pull-requests/#setting-up
- To run the behat tests you also need a working mysql locally:
  - Install the packages if you don't have them. In linux: `sudo apt install jq mysql-server mysql-client php-mysql`
  - `composer install && composer behat` => it didn't setup the database for me (mysql 8) so I did;
    - `sudo mysql -u root`
    - `CREATE DATABASE IF NOT EXISTS `wp_cli_test`;`
    - `CREATE USER 'wp_cli_test'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password1';`
    - `GRANT ALL PRIVILEGES ON wp_cli_test.* TO "wp_cli_test"@"localhost"`

## How to test

- Check out this PR locally and cd to the directory.
- `composer install`
- Create a new directory called `foo-theme` and create a `theme.json` file within with the following contents:

```json
      {
        "version": "1",
        "settings": {
          "color": {
            "palette": [
              { "slug": "black", "color": "#000000", "name": "Black" }
            ]
          }
        }
      }
```

- Run `vendor/bin/wp i18n make-pot foo-theme`
- Verify that there's a `foo-theme/foo-theme.pot` file and that the "Black" string is present as well as its context "Color name". You should see something like:

```
msgctxt "Color name"
msgid "Black"
msgstr ""
```

## Stress test the network connection

Switch to offline and verify that the file is still generated properly using the local file. Upon executing the command, you should see the following message:

```
PHP Warning:  file_get_contents(): php_network_getaddresses: getaddrinfo failed: Temporary failure in name resolution in /home/andres/src/i18n-command/src/ThemeJsonExtractor.php on line 106
PHP Warning:  file_get_contents(https://raw.githubusercontent.com/WordPress/gutenberg/wp/trunk/lib/experimental-i18n-theme.json): failed to open stream: php_network_getaddresses: getaddrinfo failed: Temporary failure in name resolution in /home/andres/src/i18n-command/src/ThemeJsonExtractor.php on line 106
Success: POT file successfully generated!
```
